pkg/disk/v1alpha1/disk.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	//clientset "github.com/openebs/maya/pkg/client/clientset/versioned"
+	clientset "github.com/openebs/maya/pkg/client/generated/clientset/internalclientset"
+)
+
+const (
+	// StorageNodePredicateKey is the key for StorageNodePredicate function.
+	filterInactivePredicateKaey = "filterInactive"
+)
+
+// KubernetesClient is the kubernetes client which will implement disk actions/behaviours.
+type KubernetesClient struct {
```

Comment:
There will be some changes to how we manage kubernetes client as we learn  more on idiomatic maya's approaches.

This PR https://github.com/openebs/maya/pull/875/commits/189177b9a350b6a40da4b07f87fb4c0ec8686381 handles kubernetes client in a particular manner. Even this will change as we progress.